src/mesh/parallel/shiftedmetricinterp.hxx

```diff
@@ -106,11 +108,18 @@ private:
   /// X-Z orthogonal to field-aligned along Y.
   Field2D zShift;
 
-  /// ZInterpolation objects for yup and ydown transformations
-  std::unique_ptr<ZInterpolation> interp_yup, interp_ydown;
+  /// Cache of interpolators for the parallel slices. Slices are stored
+  /// in the following order:
+  ///     {+1, ..., +n, -1, ..., -n}
+  /// parallel_slice_interpolator[i] stores interpolator for slice i+1
+  /// parallel_slice_interpolator[n + i] stores offset -(i+1)
+  /// where i goes from 0 to (n-1), with n the number of y guard cells
+  std::vector<std::unique_ptr<ZInterpolation>> parallel_slice_interpolators;
 
   /// ZInterpolation objects for shifting to and from field-aligned coordinates
   std::unique_ptr<ZInterpolation> interp_to_aligned, interp_from_aligned;
+
+  Mesh& mesh;
```

Comment:
Could store ` const std::size_t ydown_index = mesh.ystart`  instead of `Mesh&` to index `parallel_slice_interpolators` above? This removes the need to include the mesh header here, although we'd still need in it the .cxx file.

For consistency, might then be nice to have `constexpr std::size_t yup_index = 0;`

Reply:
The `ParallelTransform` base class already has a `Mesh& mesh` member, so I didn't need to add it to `ShiftedMetricInterp` at all... I think it's still nicer to add `yup_index` and `ydown_index` members as you suggested, it's probably nicer to have a `const size_t` than a `Mesh` member variable in the hopefully-inlined `getWeightsFor*` methods.